Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -94,6 +101,10 @@ endRideTransaction ::
   Id Merchant ->
   m ()
 endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId newFareParams thresholdConfig merchantId = do
+  booking <- QRB.findById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
```

Comment:
no need to fetch booking again, you can pass it from external code:

`endRideTransaction :: Id DP.Driver -> SRB.Booking -> DRide.Ride -> ...`

Reply:
Done